pkg/actors/actors.go

```diff
@@ -192,12 +192,13 @@ func (a *actorsRuntime) Init() error {
 
 	if len(a.config.HostedActorTypes) > 0 {
 		if a.store == nil {
-			log.Warn("actors: state store must be present to initialize the actor runtime")
-		} else {
-			features := a.store.Features()
-			if !state.FeatureETag.IsPresent(features) || !state.FeatureTransactional.IsPresent(features) {
-				return errors.New(incompatibleStateStore)
-			}
+			// If we have hosted actors and no store, we can't initialize the actor runtime
+			return fmt.Errorf("hosted actors: state store must be present to initialize the actor runtime")
+		}
+
+		features := a.store.Features()
+		if !state.FeatureETag.IsPresent(features) || !state.FeatureTransactional.IsPresent(features) {
+			return errors.New(incompatibleStateStore)
```

Comment:
These seem pretty easy to unit test, are there tests for this?

Reply:
The E2E tests cover this for sure -- there is an exclusion of the `actortestclient` app-id in the actor state storage component (I just added a commit with comments in each actor state store YAML file to make a note of that) to ensure that the client-only app used by `test_actor_features` is able to invoke actors in the the other app being deployed for that test without a state store. I don't know if we have the plumbing to fake having hosted actors in the unit test but I can look.

Reply:
Yeah, I know the e2e tests cover them, but the more places we can enforce the appropriate behavior the better. And the `NewActors` method is quite literally just this if statement, so it shouldn't be too bad to cover.

Reply:
Added unit tests in 9bb68e0ef5feb